jax/numpy/linalg.py

```diff
@@ -136,6 +136,32 @@ def eigvalsh(a, UPLO='L'):
   return w
 
 
+@_wraps(onp.linalg.pinv)
+def pinv(a, rcond=None):
+  # ported from https://github.com/numpy/numpy/blob/v1.17.0/numpy/linalg/linalg.py#L1890-L1979
+  a = np.conj(a)
+  if rcond is None:
+      if np.ndim(a) < 2:
+        num_rows = num_cols = np.shape(a)[-1]
+      else:
+        num_rows = np.shape(a)[-2]
+        num_cols = np.shape(a)[-1]
+      max_rows_cols = np.max(num_rows, num_cols)
```

Comment:
I  would suggest just writing `max_row_cols = max(a.shape[-2:])` (the Python builtin max, not numpy's max). That will work on arrays of any shape.